File: util/src/main/java/dev/openrs2/util/StringUtils.java
package dev.openrs2.util;

import com.google.common.base.Preconditions;

public final class StringUtils {
	public static String indefiniteArticle(String str) {
		Preconditions.checkArgument(!str.isEmpty());

		var first = Character.toLowerCase(str.charAt(0));
		if (first == 'a' || first == 'e' || first == 'i' || first == 'o' || first == 'u') {
			return "an";
		} else {
			return "a";
		}
	}

	public static String capitalize(String str) {
		if (str.isEmpty()) {
			return str;
		}

		var first = Character.toUpperCase(str.charAt(0));
		return first + str.substring(1);
	}

	private StringUtils() {
		/* empty */
	}
}

File: util/src/main/java/dev/openrs2/util/StringUtils.kt
package dev.openrs2.util

import dev.openrs2.util.charset.Cp1252Charset

fun CharSequence.indefiniteArticle(): String {
    require(isNotEmpty())

    return when (first().toLowerCase()) {
        'a', 'e', 'i', 'o', 'u' -> "an"
        else -> "a"
    }
}

fun CharSequence.krHashCode(): Int {
    var hash = 0
    for (c in this) {
        hash = ((hash shl 5) - hash) + Cp1252Charset.encode(c)
    }
    return hash
}
